web/modules/custom/media_videotool/src/Form/VideoToolMediaLibraryAddForm.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\media_videotool\Form;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\media\MediaTypeInterface;
+use Drupal\media_library\Form\AddFormBase;
+use function Safe\preg_match;
+
+/**
+ * Creates a form to create VideoTool media entities from within Media Library.
+ */
+class VideoToolMediaLibraryAddForm extends AddFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'media_videotool_media_library_add';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function buildInputElement(array $form, FormStateInterface $form_state): array {
+    $media_type = $this->getMediaType($form_state);
+
+    $form['container'] = [
+      '#type' => 'container',
+      '#title' => $this->t('Add @type', [
+        '@type' => $media_type->label(),
+      ]),
+    ];
+
+    $form['container']['url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Add @type video via URL', [
+        '@type' => $media_type->label(),
+      ]),
+      '#description' => $this->t('VideoTool URL example: https://media.videotool.dk/?vn=557_2023103014511477700668916683'),
+      '#required' => TRUE,
+      '#attributes' => [
+        'placeholder' => 'https://',
+      ],
+    ];
+
+    $form['container']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Add'),
+      '#button_type' => 'primary',
+      '#submit' => ['::addButtonSubmit'],
+      '#ajax' => [
+        'callback' => '::updateFormCallback',
+        'wrapper' => 'media-library-wrapper',
+        'url' => Url::fromRoute('media_library.ui'),
+        'options' => [
+          'query' => $this->getMediaLibraryState($form_state)->all() + [
+            FormBuilderInterface::AJAX_FORM_REQUEST => TRUE,
+          ],
+        ],
+      ],
+    ];
```

Comment:
Consider writing a comment of what the ajax submit functionality is validating here